kernel/src/table_properties.rs

```diff
@@ -0,0 +1,373 @@
+    /// equal to the duration specified by this property, the snapshot considers it expired and
+    /// ignores it. The SetTransaction identifier is used when making the writes idempotent.
+    #[serde(rename = "delta.setTransactionRetentionDuration")]
+    #[serde(deserialize_with = "deserialize_interval")]
+    pub set_transaction_retention_duration: Option<Duration>,
+
+    /// The target file size in bytes or higher units for file tuning. For example, 104857600
+    /// (bytes) or 100mb.
+    ///
+    /// TODO: kernel doesn't govern file writes? should we pass this through just with a note that
+    /// we don't pay attention to it? Scenario: engine calls
+    /// snapshot.table_properties().target_file_size in order to know how big _it_ should write
+    /// parquet files.
+    #[serde(rename = "delta.targetFileSize")]
+    #[serde(deserialize_with = "deserialize_pos_int")]
+    pub target_file_size: Option<u64>,
+
+    /// The target file size in bytes or higher units for file tuning. For example, 104857600
+    /// (bytes) or 100mb.
+    ///
+    /// See TODO above
+    #[serde(rename = "delta.tuneFileSizesForRewrites")]
+    #[serde(deserialize_with = "deserialize_bool")]
+    pub tune_file_sizes_for_rewrites: Option<bool>,
+
+    /// 'classic' for classic Delta Lake checkpoints. 'v2' for v2 checkpoints.
+    #[serde(rename = "delta.checkpointPolicy")]
+    #[serde(deserialize_with = "deserialize_option")]
+    pub checkpoint_policy: Option<CheckpointPolicy>,
+
+    /// TODO
+    #[serde(rename = "delta.enableRowTracking")]
+    #[serde(deserialize_with = "deserialize_bool")]
+    pub enable_row_tracking: Option<bool>,
+}
+
+impl TableProperties {
+    pub(crate) fn new(config_map: &HashMap<String, String>) -> DeltaResult<Self> {
+        let deserializer = StringMapDeserializer::new(config_map);
+        // FIXME error
+        TableProperties::deserialize(deserializer).map_err(|e| crate::Error::Generic(e.to_string()))
+    }
+
+    pub(crate) fn get_column_mapping_mode(&self) -> ColumnMappingMode {
+        self.column_mapping_mode.unwrap_or(ColumnMappingMode::None)
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Clone, Deserialize)]
+#[serde(try_from = "String")]
+pub enum DataSkippingNumIndexedCols {
+    AllColumns,
+    NumColumns(u64),
+}
+
+impl TryFrom<String> for DataSkippingNumIndexedCols {
+    type Error = String;
+
+    fn try_from(value: String) -> Result<Self, Self::Error> {
+        let num: i64 = value
+            .parse()
+            .map_err(|_| format!("couldn't parse {value} as i64"))?; // FIXME
+        match num {
+            -1 => Ok(DataSkippingNumIndexedCols::AllColumns),
+            x if x > -1 => Ok(DataSkippingNumIndexedCols::NumColumns(x as u64)),
+            _ => Err(format!("Invalid value: {}", value)),
+        }
+    }
+}
+
+/// The isolation level applied during transaction
+#[derive(Deserialize, Debug, Copy, Clone, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub enum IsolationLevel {
```

Comment:
Part of me wonders if we should defer defining table properties until we actually support them, and rely on the (to be added) unrecognized table property handling for the rest? It would simplify this PR quite a bit, and avoid checking in a bunch of dead code.

It's good to have a complete set of property flavors (bool, int, custom enum, etc) -- but we do need to recognize the v2 checkpoint table property so that should cover the custom enum case.

Reply:
yep - I'm for now including these other types in hopes of unblocking some delta-rs work since I think they would ideally want to leverage this to displace their own table properties parsing.